packages/govuk-frontend/src/govuk/macros/logo.njk

```diff
@@ -0,0 +1,77 @@
+  {%- endset %}
+
+  {#- The GOV.UK logotype with raised and coloured dot. This is only intended for
+  use alongide the Tudor Crown, so don't bother with offsetting for the
+  St. Edward's Crown. #}
+  {%- set _dotLogotype %}
+    <circle class="govuk-logo__dot" cx="113.6" cy="18" r="3.7"/>
+    <path class="govuk-logo__logotype" d="M47.4,18.1c0,.9.1,1.8.3,2.7s.6,1.6,1.1,2.3c.5.7,1.1,1.2,1.8,1.6.7.4,1.6.6,2.6.6s1.8-.2,2.4-.5c.7-.3,1.2-.7,1.6-1.2.4-.5.7-1,.8-1.5s.2-1,.2-1.5v-.2h-5.5v-3.3s9.8,0,9.8,0v12h-3.8v-2.7c-.3.4-.6.8-1,1.2-.4.4-.9.6-1.4.9-.5.2-1.1.4-1.7.6s-1.2.2-1.9.2c-1.6,0-3-.3-4.2-.9-1.2-.6-2.3-1.4-3.1-2.4s-1.5-2.2-1.9-3.5-.7-2.8-.7-4.4.2-3,.7-4.4c.5-1.4,1.2-2.5,2.1-3.5s2-1.8,3.3-2.4,2.7-.9,4.3-.9,2,.1,2.9.4c.9.2,1.8.6,2.5,1,.8.4,1.4,1,2,1.6.6.6,1.1,1.3,1.4,2.1l-3.8,2.2c-.2-.5-.5-.9-.8-1.3-.3-.4-.7-.7-1.1-1-.4-.3-.9-.5-1.4-.7s-1.1-.2-1.8-.2c-1,0-1.9.2-2.6.6-.7.4-1.3.9-1.8,1.6s-.8,1.4-1.1,2.3c-.2.9-.3,1.8-.3,2.7h0c0,.1,0,.1,0,.1ZM76.5,6.9c1.6,0,3.1.3,4.3.9s2.4,1.4,3.3,2.4,1.6,2.2,2,3.5.7,2.8.7,4.4-.2,3-.7,4.4c-.5,1.4-1.2,2.5-2,3.5s-2,1.8-3.3,2.4-2.7.9-4.3.9-3.1-.3-4.3-.9-2.4-1.4-3.2-2.4-1.6-2.2-2-3.5-.7-2.8-.7-4.4.2-3,.7-4.4,1.2-2.5,2-3.5,2-1.8,3.2-2.4,2.7-.9,4.3-.9ZM76.5,25.2c.9,0,1.8-.2,2.5-.5s1.3-.9,1.8-1.5.9-1.4,1.1-2.2c.2-.9.4-1.8.4-2.8h0c0-1.1-.1-2.1-.4-3s-.6-1.6-1.1-2.3c-.5-.6-1.1-1.1-1.8-1.5s-1.6-.5-2.5-.5-1.8.2-2.5.5c-.7.4-1.3.9-1.8,1.5s-.9,1.4-1.1,2.3-.4,1.8-.4,2.8h0c0,1.2.1,2.1.4,3,.2.9.6,1.6,1.1,2.2.5.6,1.1,1.1,1.8,1.5.7.4,1.6.5,2.5.5ZM94.6,29l-6.1-22h4.9l4.2,16.5h.2l4.1-16.5h4.9l-6.1,22h-6ZM131.5,25.2c.6,0,1.2,0,1.8-.3s1-.5,1.4-.9c.4-.4.7-.9.9-1.4.2-.6.4-1.3.4-2V7h4.3v14.2c0,1.2-.2,2.3-.7,3.3-.4,1-1.1,1.8-1.8,2.5s-1.7,1.2-2.8,1.6c-1.1.4-2.3.6-3.5.6s-2.5-.2-3.5-.6c-1.1-.4-2-.9-2.8-1.6s-1.4-1.5-1.8-2.5c-.4-1-.7-2.1-.7-3.3V7s4.3,0,4.3,0v13.6c0,.8.1,1.5.4,2,.2.6.6,1.1,1,1.4.4.4.9.7,1.4.9s1.1.3,1.8.3h0ZM144.2,7h4.4v9.6s7.8-9.6,7.8-9.6h5.4l-7.5,8.8,8,13.2h-5.1l-5.7-9.9-2.8,3.1v6.7s-4.4,0-4.4,0V7Z"/>
+  {%- endset %}
+
+  {#- The SVG needs `focusable="false"` so that Internet Explorer does not treat
+  it as an interactive element - without this it will be 'focusable' when using
+  the keyboard to navigate. #}
+  <svg
+    focusable="false"
+    role="img"
+    class="govuk-logo {%- if params.classes %} {{ params.classes }}{% endif %}"
+    xmlns="http://www.w3.org/2000/svg"
+    viewBox="0 0 {{ svgWidth }} 30"
+    height="30"
+    width="{{ svgWidth }}"
+    aria-label="GOV.UK"
+    {{- govukAttributes(params.attributes) }}
+  >
+    <title>GOV.UK</title>
```

Comment:
@selfthinker With the presence of an `aria-label` in the `<svg>` element, is that `<title>` redundant or does it cover the shortcomings of some browsers or assistive tech when it comes to the alternative text of SVGs? 

Not a blocker to merge this PR as it's what we have currently, but thought it may be something we could tidy up or document (depending on if needed or not) at some point.

Reply:
I've gone ahead and made the `aria-label` and `<title>` both optional.

That does now mean a logo can have an `img` role but no accessible name, however. We might also want to switch the role to `presentation` if there's no label configured.

Reply:
Is it possible to make the attributes dependent on each other?

If yes, ideally we'd want this if an SVG is **meaningful** (and does not include any `<text>` nodes that we might want to read instead):

```html
<svg aria-labelledby="alt-id" role="img" [...]>
  <title id="alt-id">Alternative text</title>
  ...
</svg>
```

And this if an SVG is **decorative**:

```html
<svg aria-hidden="true" [...]>
  <title>Alternative text</title>
  ...
</svg>
```

We could also add `role="presentation"` if the image is decorative, but that has the same effect as `aria-hidden="true"` and we won't need both. But if you prefer, you could use `role="presentation"` instead of `aria-hidden="true"`.

Reply:
Toggling the value of `role` feels like the simpler approach to me, that way it can be tied directly into the presence or absence of the ARIA label, rather than needing `aria-hidden` to also be added and removed.

The difference is pretty academic though, given we're both implementor and user.

Reply:
I've updated the PR so that the value of `role` is toggled based on the presence of an value for ARIA label (the `ariaLabelText` param).